docs/keycloak.rst

```diff
@@ -0,0 +1,238 @@
+   :width: 600px
+   :align: center
+
+.. image:: _static/apache1.png
+   :alt: apache-zooproject Client creation
+   :width: 600px
+   :align: center
+
+.. image:: _static/apache2.png
+   :alt: apache-zooproject Client creation
+   :width: 600px
+   :align: center
+
+- For each client:
+- After creation, go to the **Credentials** tab.
+- Copy and store the **Client Secret** for each client.
+
+.. image:: _static/webapp-secret.png
+   :alt: Copy client secret
+   :width: 600px
+   :align: center
+
+.. image:: _static/apache-secret.png
+   :alt: Copy client secret
+   :width: 600px
+   :align: center
+
+4. To create a test user:
+   - Go to **Users** → **Add user**
+
+.. image:: _static/user.png
+   :alt: Create User
+   :width: 600px
+   :align: center
+
+- After saving, go to the **Credentials** tab
+- Set a password (e.g., `test123`) and save
+
+.. image:: _static/pass.png
+   :alt: Create User Password
+   :width: 600px
+   :align: center
+
+
+Nuxt Environment Configuration
+------------------------------
+
+1. Open the file `docker/nuxt-client/env_nuxt.sample`
+
+2. Ensure that your local IP address is correctly specified in the environment variable configuration within this file.
+
+::
+
+    NUXT_OIDC_ISSUER=http://your-ip-address:8099/realms/zooproject
+    NUXT_OIDC_CLIENT_ID=webapp-zooproject
+    NUXT_OIDC_CLIENT_SECRET=<webapp-zooproject client secret>
+    NUXT_ZOO_BASEURL=http://your-ip-address
+    NUXT_BASE_URL=http://your-ip-address:3000
+    AUTH_ORIGIN=http://your-ip-address:3000
+    NEXTAUTH_URL=http://your-ip-address:3000
+
+
+Apache OIDC Configuration
+-------------------------
+
+1. Open `docker/nuxt-client/default.conf`
+
+2. Update the following block with details from the `apache-zooproject` client:
+
+::
+
+    OIDCProviderMetadataURL http://your-ip-address:8099/realms/zooproject/.well-known/openid-configuration
+    OIDCRedirectURI /
+    OIDCClientID apache-zooproject
+    OIDCClientSecret <apache-zooproject client secret>
+    OIDCCryptoPassphrase <apache-zooproject client secret>
+    OIDCRemoteUserClaim email
+    OIDCScope "openid email"
+    OIDCProviderTokenEndpointAuth client_secret_basic
+    OIDCOAuthVerifyJwksUri http://your-ip-address:8099/realms/zooproject/protocol/openid-connect/certs
+
+Configuration Files
+-------------------
+
+**main.cfg** (required headers)
+
+.. code-block:: ini
+
+  [headers]
+  Access-Control-Allow-Origin=*
+  Access-Control-Allow-Methods=GET, POST, PUT, PATCH, OPTIONS, DELETE, HEAD
+  Access-Control-Allow-Headers=Content-Type, Accept, Authorization, Origin
+
+**oas.cfg** (check rootHost/rootUrl)
+
+- Make sure they point to the IP.
+
+
+Build and Launch Nuxt Client
+----------------------------
+
+To build and run the Nuxt client:
+
+::
+
+    docker compose -f docker-compose.yml --project-directory . build nuxtclient --no-cache
+    docker compose -f docker-compose.yml --project-directory . up -d
+
+Once completed, open the Nuxt client and Login at:
+
+http://192.168.1.6:3000/
```

Comment:
Maybe here we can refer to `your-ip-address` rather than usnig a specific IP address.